datadog-profiling/src/internal/profile/mod.rs

```diff
@@ -713,8 +753,8 @@ impl Profile {
             Some(sample_types) => sample_types
                 .iter()
                 .map(|sample_type| ValueType {
-                    r#type: profile.intern(&sample_type.typ),
-                    unit: profile.intern(&sample_type.unit),
+                    r#type: Record::from(profile.intern(&sample_type.typ)),
```

Comment:
would `.into()` also work?

Reply:
Yeah but I prefer `from` over `into` in many cases because:
 1. Auto-completion works.
 2. It's more clear what it's being converted into.
 
But weirdly, we cannot implement `From<Record<T, ...>> for T` because of a quirk in Rust's type system relating to orphan rules. So we can do `From<T> for Record<T,...>` but can't do it the other way. That's why `Record.value` is `pub`. I should probably document that one.